File: cpp/2022/prelim/p07.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    double x, y, z;
    cin >> x >> y >> z;
    if (x > y) swap(x, y);
    if (y > z) swap(y, z);
    if (x > y) swap(x, y);
    if (x + y <= z) {
        cout << "Invalid";
        return 0;
    }
    if (x == z) cout << "Equilateral Triangle";
    else if (x == y || y == z) cout << "Isosceles Triangle";
    else cout << "Scalene Triangle";
    const double s = (x + y + z) / 2;
    cout << static_cast<int>(round(sqrt(s * (s - x) * (s - y) * (s - z))));
}

File: java/2022/prelim/P07.java
import java.util.Arrays;
import java.util.Scanner;

public class P07 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double[] array = new double[3];
        array[0] = scanner.nextDouble();
        array[1] = scanner.nextDouble();
        array[2] = scanner.nextDouble();
        Arrays.sort(array);
        if (array[0] + array[1] <= array[2]) {
            System.out.println("Invalid");
            return;
        }
        if (array[0] == array[2]) System.out.println("Equilateral Triangle");
        else if (array[0] == array[1] || array[1] == array[2]) System.out.println("Isosceles Triangle");
        else System.out.println("Scalene Triangle");
        double s = (array[0] + array[1] + array[2]) / 2;
        System.out.println((int) Math.round(Math.sqrt(s * (s - array[0]) * (s - array[1]) * (s - array[2]))));
    }
}
